### PR TITLE
Log improvements

### PR DESCRIPTION
**Pull request by varshavaradarajan**
* Improved logs to differenciate secure environment variable errors when cipher is invalid, encrypted value is invalid and unknown errors.
* Added logs in BackupService to provide info about how long each backup took.
* Logging the exception that is caught while trying to create artifact zip file.

@arvindsv, is this what you had in mind for backups and artifact?

### CODE REVIEW

File: server/src/com/thoughtworks/go/server/service/BackupService.java [hunk 1]
```diff
@@ -116,14 +117,26 @@ public ServerBackup startBackup(Username username, HttpLocalizedOperationResult
...
                 backupConfig(destDir);
+                LOGGER.info(String.format("Finished backing up config directory. Took {%s} ms.", System.currentTimeMillis() - start));
+                LOGGER.info(String.format("Backing up config.git repository"));
+                start = System.currentTimeMillis();
                 configRepository.doLocked(new VoidThrowingFn<IOException>() {
                     @Override public void run() throws IOException {
                         backupConfigRepository(destDir);
                     }
                 });
+                LOGGER.info(String.format("Finished backing up config directory. Took {%s} ms.", System.currentTimeMillis() - start));
```
Comment (zabil) [inline server/src/com/thoughtworks/go/server/service/BackupService.java]: Spring has this for measuring time http://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/StopWatch.html use that if spring classes are available in the classpath

File: server/src/com/thoughtworks/go/server/cache/ArtifactCache.java [hunk 2]
```diff
@@ -24,11 +24,13 @@
 import java.util.concurrent.ConcurrentSkipListSet;
 
 import com.thoughtworks.go.server.service.ArtifactsDirHolder;
+import org.apache.log4j.Logger;
```
Comment (zabil) [inline server/src/com/thoughtworks/go/server/cache/ArtifactCache.java]: Use slf4j

File: server/src/com/thoughtworks/go/server/cache/ArtifactCache.java [hunk 3]
```diff
@@ -74,6 +76,7 @@ public void run() {
                     try {
                         createCachedFile(artifactLocation);
                     } catch (Exception e) {
+                        LOG.error(String.format("An error occurred while trying to create the artifact zip file of directory '%s'", artifactLocation), e);
```
Comment (zabil) [inline server/src/com/thoughtworks/go/server/cache/ArtifactCache.java]: with slf4j use paramerized logging http://slf4j.org/faq.html#logging_performance

File: server/src/com/thoughtworks/go/server/cache/ArtifactCache.java (hunk 2 above)
Comment (zabil) [inline server/src/com/thoughtworks/go/server/cache/ArtifactCache.java]: Also refer comments on https://github.com/gocd/gocd/pull/2452
For testing the log messages.

**Comment by arvindsv** [comment]
@varshavaradarajan Yes, that looks ok. Not sure about the use of org.springframework.util.StopWatch. :) I see that you've not been scarred by Spring yet.

**Pull request merged**